Common/CostFunctions/itkAdvancedImageToImageMetric.hxx

```diff
@@ -368,10 +345,10 @@ AdvancedImageToImageMetric<TFixedImage, TMovingImage>::CheckForBSplineInterpolat
     using RayCastInterpolatorType =
       itk::AdvancedRayCastInterpolateImageFunction<MovingImageType, CoordinateRepresentationType>;
     const bool interpolatorIsRayCast =
-      dynamic_cast<RayCastInterpolatorType *>(this->m_Interpolator.GetPointer()) != nullptr;
+      dynamic_cast<RayCastInterpolatorType *>(Superclass::m_Interpolator.GetPointer()) != nullptr;
 
-    if (!this->m_InterpolatorIsBSpline && !this->m_InterpolatorIsBSplineFloat &&
-        !this->m_InterpolatorIsReducedBSpline && !this->m_InterpolatorIsLinear && !interpolatorIsRayCast)
+    if (!m_BSplineInterpolator && !m_BSplineInterpolatorFloat && !m_ReducedBSplineInterpolator &&
+        !m_LinearInterpolator && !interpolatorIsRayCast)
     {
       using CentralDifferenceGradientFilterType = GradientImageFilter<TMovingImage, RealType, RealType>;
```

Comment:
@mstaring @stefanklein Looking at [elastix/Components/Interpolators](https://github.com/SuperElastix/elastix/tree/43b5c54e2486ec2517c8abd92fa2b69e9bb55975/Components/Interpolators), it appears to me that elastix supports 6 different interpolator types: BSpline, BSpline Float, ReducedBSpline, Linear, NearestNeighbor, RayCast. Only NearestNeighbor uses the image produced by ITK's `GradientImageFilter`, right?

If so, then why not to just test on a single condition, `if (interpolator is NearestNeighbor)` ?